coordinator/Dockerfile

```diff
@@ -1,20 +1,38 @@
-FROM openjdk:22-ea-17-slim-bookworm
+# BUILDER image
+FROM eclipse-temurin:21-jdk-noble AS builder
 
-RUN apt-get update \
-	&& apt-get install curl -y \
-  && apt-get install unzip -y \
-	&& apt-get clean \
-	&& rm -rf /var/lib/apt/lists/*
+RUN mkdir -p /libs
+COPY --from=libs ./** /libs
+WORKDIR /libs
+
+RUN mkdir -p unpacked-blob-compressor unpacked-blob-shnarf-calculator
+RUN bcJar=$(set -- blob-compressor*.jar; echo $1) && \
+    bscJar=$(set -- blob-shnarf-calculator*.jar; echo $1) && \
+    cd unpacked-blob-compressor/ && jar -xf ../$bcJar && cd .. && \
+    cd unpacked-blob-shnarf-calculator/ && jar -xf ../$bscJar && cd .. && \
+    rm -rf unpacked-blob-compressor/darwin-** unpacked-blob-shnarf-calculator/darwin-** && \
+    case $(uname -m) in \
+    x86_64) \
+      rm -rf unpacked-blob-compressor/linux-aarch64/; \
+      rm -rf unpacked-blob-shnarf-calculator/linux-aarch64/; \
+    ;; \
+    aarch64) \
+      rm -rf unpacked-blob-compressor/linux-x86-64/; \
+      rm -rf unpacked-blob-shnarf-calculator/linux-x86-64/; \
+    ;; \
+    esac && \
+    jar -cf $bcJar -C unpacked-blob-compressor . && \
+    jar -cf $bscJar -C unpacked-blob-shnarf-calculator .
+RUN rm -rf unpacked-blob-compressor unpacked-blob-shnarf-calculator
+
+# FINAL image
+FROM eclipse-temurin:21-jre-noble
```

Comment:
We can't use `eclipse-temurin:21.0.4_7-jre-alpine` or `eclipse-temurin:21-jre-alpine` here because we'd get the following error at runtime:
```
2024-11-22 14:55:03 Native library (root/.cache/JNA/temp/jna3653017674932247023.tmp) not found in resource path (libs/coordinator.jar)
2024-11-22 14:55:03     at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:325)
2024-11-22 14:55:03     at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:481)
2024-11-22 14:55:03     at com.sun.jna.Library$Handler.<init>(Library.java:197)
2024-11-22 14:55:03     at com.sun.jna.Native.load(Native.java:618)
2024-11-22 14:55:03     at com.sun.jna.Native.load(Native.java:592)
2024-11-22 14:55:03     at net.consensys.linea.blob.GoNativeBlobCompressorFactory$Companion.getInstance(GoNativeBlobCompressor.kt:118)
```
which could be caused by the known issue that Alpine is using `muslibc` instead of `glibc` as mentioned [here](https://github.com/iegomez/mosquitto-go-auth/issues/20#issuecomment-497017101), I tried adding the apk package `gcompat` as suggested by others but no help, or it might be related to how we build the jna libs, but instead of spending more time into this issue, I'd recommend to keep using `eclipse-temurin:21-jre-noble` for now, and if its really the requirement for us to support alpine image for coordinator, we could add a ticket to following this up in greater length